client.cpp

```diff
@@ -878,6 +878,130 @@ bool verify_client::finished(void)
 
 ///////////////////////////////////////////////////////////////////////////
 
+crc_verify_client::verify_request::verify_request(request_type type,
+                                              unsigned int size,
+                                              struct timeval* sent_time,
+                                              unsigned int keys,
+                                              const char *key,
+                                              unsigned int key_len) :
+        client::request(type, size, sent_time, keys),
+        m_key(NULL), m_key_len(0)
+{
+    m_key_len = key_len;
+    m_key = (char *)malloc(key_len);
+    memcpy(m_key, key, m_key_len);
+}
+
+crc_verify_client::verify_request::~verify_request(void)
+{
+    if (m_key != NULL) {
+        free((void *) m_key);
+        m_key = NULL;
+    }
+}
+
+crc_verify_client::crc_verify_client(struct event_base *event_base,
+                             benchmark_config *config,
+                             abstract_protocol *protocol,
+                             object_generator *obj_gen) : client(event_base, config, protocol, obj_gen),
+                                                          m_finished(false), m_verified_keys(0), m_errors(0)
+{
+    m_protocol->set_keep_value(true);
+    m_obj_gen->set_key_range(config->key_minimum, config->key_maximum);
+}
+
+unsigned long long int crc_verify_client::get_verified_keys(void)
+{
+    return m_verified_keys;
+}
+
+unsigned long long int crc_verify_client::get_errors(void)
+{
+    return m_errors;
+}
+
+void crc_verify_client::create_request(struct timeval timestamp)
+{
+    // Prepare a GET request that will be compared against a previous
+    // SET request.
+    unsigned int cmd_size;
+    int iter = obj_iter_type(m_config, 2);
+    unsigned int keylen;
+    const char *key = m_obj_gen->get_key(iter, &keylen);
+    assert(key != NULL);
+    assert(keylen > 0);
+
+    benchmark_debug_log("CRC verify: GET key=[%.*s]\n", keylen, key);
+    cmd_size = m_protocol->write_command_get(key, keylen, m_config->data_offset);
+    m_pipeline.push(new crc_verify_client::verify_request(rt_get, cmd_size, &timestamp, 1, key, keylen));
+}
+
+void crc_verify_client::handle_response(struct timeval timestamp, request *request, protocol_response *response)
+{
+    unsigned int rvalue_len;
+    const char *rvalue = response->get_value(&rvalue_len);
+    verify_request *vr = static_cast<verify_request *>(request);
+
+    assert(vr->m_type == rt_get);
```

Comment:
what code handles the response of a SET command?

Reply:
No code, currently this client only generates GET commands, hence the assert.

Reply:
so, you use regular client to issue the SET commands and verify client to issue GET commands?

Reply:
yes